sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/OptimizeInSuite.scala

```diff
@@ -238,4 +238,34 @@ class OptimizeInSuite extends PlanTest {
 
     comparePlans(optimized, correctAnswer)
   }
+
+  test("OptimizedIn test: optimize range compare") {
+    val originalQuery = testRelation
+      .select('a)
+      .where('a in(4, 1, 2, 3, 3, 5))
+
+    val optimized = Optimize.execute(originalQuery.analyze)
+
+    val expected = testRelation
+      .select('a)
+      .where('a >= 1 && 'a <= 5)
+      .analyze
+
+    comparePlans(optimized, expected)
+  }
+
+  test("OptimizedIn test: do not optimize range compare if non-continuous") {
+    val originalQuery = testRelation
+      .select('a)
+      .where('a in(1, 2, 3, 3, 5))
+
+    val optimized = Optimize.execute(originalQuery.analyze)
+
+    val expected = testRelation
+      .select('a)
+      .where('a in(1, 2, 3, 5))
```

Comment:
might be optimized into 
> 'a >= 1 && 'a <= 5 && 'a != 4 
? 
May not make a lot of sense for this simple use case, but for a longer IN list this would be nice to be covered as well.

Reply:
agreed. though I think something like a >=1 && a <= 3 and a == 5 would be more simple to implement.